### DISCUSSION TITLE
Conditional fields lose their value when not present

### DISCUSSION THREAD

**Discussion by dan-gamble** [edmundhung/conform#455] [Q&A]
### Describe the bug and the expected behavior

If an element disappears from the form it's value is then wiped from the `form.value`.

I'd built quite a deeply nested tab-based form and discovered that when React removes an element from the DOM then `conform` also removes that value from `form.value`. I'd expect the data to stay as it was in the last state before the element was removed from the DOM.

### Conform version

v1.0.1

### Steps to Reproduce the Bug or Issue

1. Go to the [sandbox](https://codesandbox.io/p/sandbox/zod-conditional-elements-krdm95?file=%2Fsrc%2FApp.tsx%3A65%2C21&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clslm7dnk00062e66x1zw4pim%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clslm7dnk00022e66x8pb5ywm%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clslm7dnk00032e664i36d1qc%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clslm7dnk00052e66qp910tvi%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clslm7dnk00022e66x8pb5ywm%2522%253A%257B%2522id%2522%253A%2522clslm7dnk00022e66x8pb5ywm%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clslmj5ba00022e667lm41sx2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A65%252C%2522startColumn%2522%253A21%252C%2522endLineNumber%2522%253A65%252C%2522endColumn%2522%253A21%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clslmj5ba00022e667lm41sx2%2522%257D%252C%2522clslm7dnk00052e66qp910tvi%2522%253A%257B%2522id%2522%253A%2522clslm7dnk00052e66qp910tvi%2522%252C%2522activeTabId%2522%253A%2522clslm94sj006g2e66pc7jb4oy%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clslm94sj006g2e66pc7jb4oy%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clslm7dnk00032e664i36d1qc%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clslm7dnk00032e664i36d1qc%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)
2. Change the dropdown to a different value
3. See that the `bike` or `car` object is removed from the `form.value`

For the sake of this demo I've used a select but this was initially discovered when I was using tabs

### What browsers are you seeing the problem on?

Chrome, Firefox, Microsoft Edge, Safari, Others

**Comment by edmundhung** [comment]
Conform does not keep any value of your form. What you get from Conform is just a value synced with the FormData API. If you want to persist the value, you need to eithe make the input visually hidden or render a hidden input yourself.

The DOM is the source of truth of your formdata as that's what will be sent to the server regardless you are using Remix route action or NextJS server action. It would be bad if a library tells your the value exists, but it is not actually sent out to the server :)
